app/src/main/java/com/islamey/hamza/wazaker/ui/CounterFragment/CounterViewModel.kt

```diff
@@ -0,0 +1,77 @@
+package com.islamey.hamza.wazaker.ui.CounterFragment
+
+import android.app.Application
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+@HiltViewModel
+class CounterViewModel @Inject constructor(
+    private val application: Application
+) : ViewModel() {
+
+    private val _counts = MutableStateFlow(0)
+    val counts: StateFlow<Int> get() = _counts
+
+    private val _totalCounts = MutableStateFlow(0)
+    val totalCounts: StateFlow<Int> get() = _totalCounts
+
+    init {
+        viewModelScope.launch {
+            _totalCounts.value = getTotalCountsFromSharedPreference()
+        }
+    }
+
+    fun incrementCounter() {
+        viewModelScope.launch {
+            _counts.value++
+            _totalCounts.value++
+            saveTotalCountsInSharedPreference()
+        }
+    }
+
+    fun resetCounter() {
+        viewModelScope.launch {
+            _counts.value = 0
+        }
+    }
+
+    fun resetTotalCounter() {
+        viewModelScope.launch {
+            _totalCounts.value = 0
+            saveTotalCountsInSharedPreference()
+        }
+    }
```

Comment:
do we need launch here ? I think no need.

Reply:
That's right